jax_md/elasticity.py

```diff
@@ -0,0 +1,526 @@
+                                [[0, 1, 1], [1, 0, 0], [1, 0, 0]]], dtype=dtype)
+  else:
+    raise AssertionError('not implemented for {} dimensions'.format(dim))
+  return strain_tensors
+
+def _convert_responses_to_elastic_constants(response_all: Array) -> Array:
+  if response_all.shape[0] == 6:
+    cxxxx = response_all[0]
+    cyyyy = response_all[1]
+    cxyxy = 0.25 * response_all[2]
+    cxxyy = 0.5  * (response_all[3] - cxxxx - cyyyy)
+    cxxxy = 0.25 * (response_all[4] - 4 * cxyxy - cxxxx)
+    cyyxy = 0.25 * (response_all[5] - 4 * cxyxy - cyyyy)
+
+    C = jnp.array(
+        [[[[cxxxx, cxxxy], [cxxxy, cxxyy]],
+          [[cxxxy, cxyxy], [cxyxy, cyyxy]]],
+         [[[cxxxy, cxyxy], [cxyxy, cyyxy]],
+          [[cxxyy, cyyxy], [cyyxy, cyyyy]]]])
+
+  elif response_all.shape[0] == 21: 
+    cxxxx = response_all[0];
+    cyyyy = response_all[1];
+    czzzz = response_all[2];
+    cyzyz = response_all[3]/4.;
+    cxzxz = response_all[4]/4.;
+    cxyxy = response_all[5]/4.;
+    cyyzz = (response_all[6] - cyyyy - czzzz)/2.;
+    cxxzz = (response_all[7] - cxxxx - czzzz)/2.;
+    cxxyy = (response_all[8] - cxxxx - cyyyy)/2.;
+    cxxyz = (response_all[9] - cxxxx - 4.*cyzyz)/4.;
+    cxxxz = (response_all[10] - cxxxx - 4.*cxzxz)/4.;
+    cxxxy = (response_all[11] - cxxxx - 4.*cxyxy)/4.;
+    cyyyz = (response_all[12] - cyyyy - 4.*cyzyz)/4.;
+    cyyxz = (response_all[13] - cyyyy - 4.*cxzxz)/4.;
+    cyyxy = (response_all[14] - cyyyy - 4.*cxyxy)/4.;
+    czzyz = (response_all[15] - czzzz - 4.*cyzyz)/4.;
+    czzxz = (response_all[16] - czzzz - 4.*cxzxz)/4.;
+    czzxy = (response_all[17] - czzzz - 4.*cxyxy)/4.;
+    cyzxz = (response_all[18] - 4.*cyzyz - 4.*cxzxz)/8.;
+    cyzxy = (response_all[19] - 4.*cyzyz - 4.*cxyxy)/8.;
+    cxzxy = (response_all[20] - 4.*cxzxz - 4.*cxyxy)/8.;
+
+    C = jnp.array(
+         [[[[cxxxx, cxxxy, cxxxz],
+            [cxxxy, cxxyy, cxxyz],
+            [cxxxz, cxxyz, cxxzz]],
+           [[cxxxy, cxyxy, cxzxy],
+            [cxyxy, cyyxy, cyzxy],
+            [cxzxy, cyzxy, czzxy]],
+           [[cxxxz, cxzxy, cxzxz],
+            [cxzxy, cyyxz, cyzxz],
+            [cxzxz, cyzxz, czzxz]]],
+          [[[cxxxy, cxyxy, cxzxy],
+            [cxyxy, cyyxy, cyzxy],
+            [cxzxy, cyzxy, czzxy]],
+           [[cxxyy, cyyxy, cyyxz],
+            [cyyxy, cyyyy, cyyyz],
+            [cyyxz, cyyyz, cyyzz]],
+           [[cxxyz, cyzxy, cyzxz],
+            [cyzxy, cyyyz, cyzyz],
+            [cyzxz, cyzyz, czzyz]]],
+          [[[cxxxz, cxzxy, cxzxz],
+            [cxzxy, cyyxz, cyzxz],
+            [cxzxz, cyzxz, czzxz]],
+           [[cxxyz, cyzxy, cyzxz],
+            [cyzxy, cyyyz, cyzyz],
+            [cyzxz, cyzyz, czzyz]],
+           [[cxxzz, czzxy, czzxz],
+            [czzxy, cyyzz, czzyz],
+            [czzxz, czzyz, czzzz]]]])
+  else:
+    raise AssertionError('incorrect shape for array of responses')
+  return C
+
+def AthermalElasticModulusTensor(energy_fn: Callable[..., Array], 
```

Comment:
To go with everything else in the library this should be lowercase, e.g. `athermal_elastic_modulus_tensor`. Also, do we need the `tensor` here? Could it just be `athermal_elastic_modulus`? Since we're in a file called elasticity it could even just be `athermal_moduli`, perhaps?

Reply:
great. going with `athermal_moduli`